app/controllers/deliveries_controller.rb

```diff
@@ -47,4 +47,24 @@ def claim
       end
     end
   end
+  
+  def update_status
+    respond_to do |format|
+      format.json do
+        if @driver = Driver.find_by(key: params[:key])
+          if @delivery = Receipt.find_by(id: params[:delivery_id])
+              if @delivery.update(status: params[:status])
+                render :json => @delivery.status
+              else
+                render :json => {error: "Invalid status"}
+              end
+          else
+            render :json => {error: "Receipt does not exist"}
```

Comment:
Might be more sensical to say "Delivery does not exist"

As hypothetically it'll only be the driver to access this endpoint, and it'll save a bit of translation work on the client-side